src/transformers/onnx/convert.py

```diff
@@ -100,11 +100,18 @@ def export_pytorch(
         `Tuple[List[str], List[str]]`: A tuple with an ordered list of the model's inputs, and the named inputs from
         the ONNX configuration.
     """
+    from ..tokenization_utils_base import PreTrainedTokenizerBase
+
+    if isinstance(preprocessor, PreTrainedTokenizerBase) and tokenizer is not None:
+        raise ValueError("You cannot provide both a tokenizer and a preprocessor to export the model.")
     if tokenizer is not None:
         warnings.warn(
             "The `tokenizer` argument is deprecated and will be removed in version 5 of Transformers. Use `preprocessor` instead.",
             FutureWarning,
         )
+        logger.warning("Overwriting the `preprocessor` argument with `tokenizer` to generate dummmy inputs.")
```

Comment:
Maybe this can be an `info` as it's more additional information and not really an error

Reply:
(warnings get displayed by default, info is displayed when users ask to have more info)